frontend/src/project/ProjectTab.tsx

```diff
@@ -0,0 +1,91 @@
+import { octokit } from '@/shared/apis/octokit';
+import {
+  GITHUB_OWNER,
+  GITHUB_REPO,
+  GITHUB_DIR_SAMPLE_SHA,
+} from '@/shared/constants/github';
+import { TreeNode } from '@/shared/interfaces/tree';
+import { buildTreeFromArray } from '@/shared/utils/tree';
+import { FileOutlined, FolderFilled } from '@ant-design/icons';
+import { Button, Layout, Skeleton } from 'antd';
+import { Content } from 'antd/es/layout/layout';
+import Sider from 'antd/es/layout/Sider';
+import { useState, useEffect } from 'react';
+import { useQuery } from 'react-query';
+
+export const ProjectTab = () => {
+  const [tree, setTree] = useState<TreeNode[]>([]);
+  const [content, setContent] = useState<string | null>('');
+
+  const { isPending, error, data } = useQuery({
+    queryKey: 'project',
+    queryFn: async () => {
+      const response = await octokit.rest.git.getTree({
+        owner: GITHUB_OWNER,
+        repo: GITHUB_REPO,
+        tree_sha: GITHUB_DIR_SAMPLE_SHA,
+        recursive: '0',
+      });
+      return response.data;
+    },
+  });
+
+  useEffect(() => {
+    if (data) {
+      setTree(buildTreeFromArray(data.tree));
+    }
+  }, [data]);
+
+  const renderTree = (nodes: TreeNode[]) => {
+    return nodes.map((node) => {
+      return (
+        <div key={node.path} className="px-2">
+          <Button
+            icon={node.type === 'tree' ? <FolderFilled /> : <FileOutlined />}
+            onClick={async (
+              e: React.MouseEvent<HTMLButtonElement, MouseEvent>,
+            ) => {
+              e.preventDefault();
+              if (node.type === 'blob') {
```

Comment:
put 'blob' string to enum

Reply:
![image](https://github.com/phhytrg/selflearn/assets/74171041/1b47f72c-cd12-4322-8dd1-386facc7a43c)